plasma/transaction.go

```diff
@@ -61,42 +59,47 @@ func (tx *Transaction) DecodeRLP(s *rlp.Stream) error {
 	confirmSigs0 := parseSig(t.Tx.Input0ConfirmSigs)
 	confirmSigs1 := parseSig(t.Tx.Input1ConfirmSigs)
 
-	tx.Input0 = NewInput(NewPosition(big.NewInt(new(big.Int).SetBytes(t.Tx.BlkNum0[:]).Int64()), uint16(new(big.Int).SetBytes(t.Tx.TxIndex0[:]).Int64()), uint8(new(big.Int).SetBytes(t.Tx.OIndex0[:]).Int64()), big.NewInt(new(big.Int).SetBytes(t.Tx.DepositNonce0[:]).Int64())),
-		t.Sigs[0], confirmSigs0)
-	tx.Input1 = NewInput(NewPosition(big.NewInt(new(big.Int).SetBytes(t.Tx.BlkNum1[:]).Int64()), uint16(new(big.Int).SetBytes(t.Tx.TxIndex1[:]).Int64()), uint8(new(big.Int).SetBytes(t.Tx.OIndex1[:]).Int64()), big.NewInt(new(big.Int).SetBytes(t.Tx.DepositNonce1[:]).Int64())),
-		t.Sigs[1], confirmSigs1)
+	tx.Inputs = append(tx.Inputs, NewInput(NewPosition(big.NewInt(new(big.Int).SetBytes(t.Tx.BlkNum0[:]).Int64()), uint16(new(big.Int).SetBytes(t.Tx.TxIndex0[:]).Int64()), uint8(new(big.Int).SetBytes(t.Tx.OIndex0[:]).Int64()), big.NewInt(new(big.Int).SetBytes(t.Tx.DepositNonce0[:]).Int64())),
+		t.Sigs[0], confirmSigs0))
+	pos := NewPosition(big.NewInt(new(big.Int).SetBytes(t.Tx.BlkNum1[:]).Int64()), uint16(new(big.Int).SetBytes(t.Tx.TxIndex1[:]).Int64()), uint8(new(big.Int).SetBytes(t.Tx.OIndex1[:]).Int64()), big.NewInt(new(big.Int).SetBytes(t.Tx.DepositNonce1[:]).Int64()))
+	if !pos.IsNilPosition() {
+		tx.Inputs = append(tx.Inputs, NewInput(pos, t.Sigs[1], confirmSigs1))
+	}
 	// set signatures if applicable
-	tx.Output0 = NewOutput(t.Tx.NewOwner0, big.NewInt(new(big.Int).SetBytes(t.Tx.Amount0[:]).Int64()))
-	tx.Output1 = NewOutput(t.Tx.NewOwner1, big.NewInt(new(big.Int).SetBytes(t.Tx.Amount1[:]).Int64()))
+	tx.Outputs = append(tx.Outputs, NewOutput(t.Tx.NewOwner0, big.NewInt(new(big.Int).SetBytes(t.Tx.Amount0[:]).Int64())))
+	if !utils.IsZeroAddress(t.Tx.NewOwner1) {
+		tx.Outputs = append(tx.Outputs, NewOutput(t.Tx.NewOwner1, big.NewInt(new(big.Int).SetBytes(t.Tx.Amount1[:]).Int64())))
+	}
 	tx.Fee = big.NewInt(new(big.Int).SetBytes(t.Tx.Fee[:]).Int64())
 
 	return nil
 }
 
 func (tx Transaction) ValidateBasic() error {
 	// validate inputs
-	if err := tx.Input0.ValidateBasic(); err != nil {
-		return fmt.Errorf("invalid first input { %s }", err)
-	}
-	if tx.Input0.Position.IsNilPosition() {
-		return fmt.Errorf("first input cannot be nil")
+	for i, input := range tx.Inputs {
```

Comment:
Maybe put 2 I/O check in here for now since we still need to be compatible with rootchain v1.0